block/base/handlers.go

```diff
@@ -111,35 +111,59 @@ func (l *BaseLane) DefaultPrepareLaneHandler() PrepareLaneHandler {
 
 // DefaultProcessLaneHandler returns a default implementation of the ProcessLaneHandler. It verifies
 // the following invariants:
-// 1. All transactions belong to this lane.
-// 2. All transactions respect the priority defined by the mempool.
-// 3. All transactions are valid respecting the verification logic of the lane.
+//  1. Transactions belonging to the lane must be contiguous from the beginning of the partial proposal.
+//  2. Transactions that do not belong to the lane must be contiguous from the end of the partial proposal.
+//  3. Transactions must be ordered respecting the priority defined by the lane (e.g. gas price).
+//  4. Transactions must be valid according to the verification logic of the lane.
 func (l *BaseLane) DefaultProcessLaneHandler() ProcessLaneHandler {
```

Comment:
Where are we checking that each lane doesn't consume more gas / bytes than the lane-limit

Reply:
this is done for you while it rebuilds the same block proposal as what was constructed in prepare. prepare is the only one that has to proactively check that it doesnt exceed lane limit, this can just optimistically update and if that fails bc its either too gas intensive or too large it gets rejected.

Reply:
https://github.com/skip-mev/block-sdk/blob/8b25b0364f1c26b46d4d6797fe2abc4e1a6c5486/block/base/abci.go#L92-L115